.github/workflows/xcframework.yml

```
@@ -0,0 +1,117 @@
+#                      -derivedDataPath ./build/derivedDataPath \
+#                      SKIP_INSTALL=NO
+#                  xcodebuild -create-xcframework \
+#                          -framework $(pwd)/build/Release-macOS.xcarchive/Products/Library/Frameworks/ElasticApm.framework \
+#                          -output ./package/ElasticApm_macOS.xcframework
+            - name: generate tvOS xcframework
+              run: |
+                xcodebuild clean archive -scheme ElasticApm_tvOS \
+                    -configuration Release \
+                    -destination 'generic/platform=tvOS' \
+                    -archivePath ./build/Release-tvOS.xcarchive \
+                    -derivedDataPath ./build/derivedDataPath \
+                    SKIP_INSTALL=NO
+                xcodebuild -create-xcframework \
+                        -framework $(pwd)/build/Release-tvOS.xcarchive/Products/Library/Frameworks/ElasticApm.framework \
+                        -output ./package/ElasticApm_tvOS.xcframework
+            - name: Package frameworks
+              id: package-frameworks
+              run: |
+                av=sed -n -E "s/public static let elasticSwiftAgentVersion: String = \"(.*)\"/\1/p" Sources/apm-agent-ios/Version.swift  | sed -e 's/^[[:space:]]*//'
+                echo "using version ${av}"
+                echo "APP_VERSION=${av}" >> "$GITHUB_ENV"
+                rm -rf "./dist"
+                mkdir -p "./dist"
+                echo $(pwd)
+                tar -czf "./dist/ElasticApm_tvOS-${av}.tar.gz" -C ./package ElasticApm_tvOS.xcframework
+#                tar -czf "./dist/ElasticApm_macOS-${APP_VERSION}.tar.gz" -C ./package ElasticApm_macOS.xcframework
+#                tar -czf "./dist/ElasticApm_iOS-${APP_VERSION}.tar.gz" -C ./package ElasticApm_iOS.xcframework
+            - name: Generate tvOS Framework provenance
+              uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d  # v1.1.2
+              env:
+                APP_VERSION: ${{ steps.package-frameworks.outputs.APP_VERSION }}
+              with:
+                subject-path: "${{ github.workspace }}/dist/ElasticApm_tvOS-${APP_VERSION}.tar.gz"
+            - name: Generate iOS Framework provenance
+              uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d  # v1.1.2
+              env:
+                APP_VERSION: ${{ steps.package-frameworks.outputs.APP_VERSION }}
+              with:
+                subject-path: "${{ github.workspace }}/build/ElasticApm_iOS-${APP_VERSION}.tar.gz"
+            - name: Generate macOS Framework provenance
+              uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d  # v1.1.2
+              env:
+                APP_VERSION: ${{ steps.package-frameworks.outputs.APP_VERSION }}
+              with:
+                subject-path: "${{ github.workspace }}/build/ElasticApm_macOS-${APP_VERSION}.tar.gz"
```

If all these files are `tar.gz` then you can use [wildcards](https://github.com/actions/attest-build-provenance#identify-subjects-by-wildcard):

```suggestion
            - name: Generate provenance
              uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d  # v1.1.2
              with:
                subject-path: "${{ github.workspace }}/**/*.tar.gz"
```

---

Will this produce a hash for each individual artifact?

---

That's my understanding